### DISCUSSION TITLE
Getting error while using readImageBlob() with inline SVG.

### DISCUSSION THREAD

**Discussion by gurpreetindybytes** [ImageMagick/ImageMagick#2333] [Help]
I am using the Imagick library to convert the inline SVG into the PNG file. I am using this library with Laravel. firstly I have initialized the instance of Imagick.`$magick = new \Imagick();`. After that when I have passed the inline SVG variable (that contains the SVG data) into the `readImageBlob($svg)` it's throwing the error "no decode delegate for this image format `' @ error/blob.c/BlobToImage/458". 
Here is my sample code :- 

$imagick = new \Imagick();
$imagick->readImageBlob($svg); // on this line I am getting **no decode delegate for this image format `' @ error/blob.c/BlobToImage/458** error
`

**Comment by snibgo** [comment]
I don't use IMagick so probably can't help. For those who do, I suggest you show complete code, and the value of $svg.

**Reply by gurpreetindybytes** [reply]
When I copy the data in my question and update it. It never displays the SVG code in my question.

**Comment by mattsrinc** [comment]
"No delegate" message means there is no library to support SVG format on the system. That can be Ghostscript, rsvg, Inkscape or built in ImageMagick.

So, you need to check the phpinfo() output for IMagick extension to see what formats it supports. And then work with your ISP to add it - recommending to use the latest ImageMagick (no version 6!). ImageMagick from version 7.0.10 (?) supports proper SVG handling, including not creating SVG circle elements for every dot in the image ...
